ledger-service/http-json/src/main/scala/com/digitalasset/http/EndpointsCompanion.scala

```diff
@@ -55,22 +64,134 @@ object EndpointsCompanion {
       case NonFatal(t) => ServerError(t.description)
     }
   }
-
+  private type ET[A] = EitherT[Future, Unauthorized, A]
   trait ParsePayload[A] {
-    def parsePayload(jwt: DecodedJwt[String]): Unauthorized \/ A
+    def parsePayload(
+        jwt: DecodedJwt[String]
+    ): Error \/ A
+  }
+
+  trait CreateFromUserToken[A] {
```

Comment:
maybe just make this a type syonym?

Reply:
or in fact maybe make a class that accepts the last 2 parameters and then have each method only take the jwt?